### PR TITLE
fix: Modal widget background issues

### LINKED ISSUE #20405: [Bug]: Modal's background is missing in deploy mode
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Modal's background is missing in deployed view

https://www.loom.com/share/48e47ec1502845019cda25e0c5e183ea

### Steps To Reproduce

1. Add a modal to the canvas
2. Deploy the app
3. Notice that the modal's background is missing

### Public Sample App

_No response_

### Issue video log

https://www.loom.com/share/48e47ec1502845019cda25e0c5e183ea

### Version

Release

### LINKED ISSUE #20434: [Bug]: Unable to change BG color of Modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Unable to change BG color of Modal 

### Steps To Reproduce

1. DnD a modal 
2. try changing the bg color

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

release

### LINKED ISSUE #20435: [Bug]: On adding a partial evaluation in the table filter, the results being fetched are incorrect
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On adding a partial evaluation in the table filter, the results being fetched are incorrect

### Steps To Reproduce

1. D&D a table widget
2. Deploy the app
3. Add a valid condition in the filter 
4. Add a partial condition (without any attributes)
5. Notice that on adding the second filter, the data which is being fetched is incorrect

### Public Sample App

_No response_

### Issue video log

https://www.loom.com/share/f051a226e2484f3db350c6e5cc882d28

### Version

Production

### LINKED ISSUE #20436: [Bug]:Color picker icon missing from Color properties in all widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Color picker icon missing when no color is selected from Color properties in all widgets in release 

### Steps To Reproduce
Release : 
Screen Shot 2023-02-07 at 4 36 52 PM

Prod:
Screen Shot 2023-02-07 at 4 36 34 PM

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by riodeuno**
## Description

The following issues were caused because of the recent changes with respect to auto height instant update.
In the recent change, we removed a few wrappers around containers which seemed unnecessary. This led to the fact that in deploy mode, these wrappers were not present. The issue with this was that, these wrappers were responsible for the modal widget's background color styling. This also led to an issue where the background color was not applied in edit mode.

To fix this, we've added a wrapper in the component and removed all styling logic from the widget. This is because, the component is responsible for the actual rendering and what users see. The widgets should act as an interface, and as a result, should not deal with styling.

Fixes #20434 
Fixes #20405 
Fixes #20436 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — DISMISSED** [review]

File: app/client/src/utils/hooks/useClickToSelectWidget.tsx [hunk 1]
```diff
@@ -10,43 +10,21 @@ import {
 } from "selectors/widgetSelectors";
 import styled from "styled-components";
 import { stopEventPropagation } from "utils/AppsmithUtils";
-import { scrollCSS } from "widgets/WidgetUtils";
+// import { scrollCSS } from "widgets/WidgetUtils";
```
Comment (marks0351) [inline app/client/src/utils/hooks/useClickToSelectWidget.tsx:13]: lets remove if not needed

**Review by marks0351 — DISMISSED** [review]

File: app/client/src/widgets/ModalWidget/widget/index.tsx [hunk 2]
```diff
@@ -260,7 +274,7 @@ export class ModalWidget extends BaseWidget<ModalWidgetProps, WidgetState> {
   getCanvasView() {
     let children = this.getChildren();
     children = this.makeModalSelectable(children);
-    children = this.showWidgetName(children, true);
+    // children = this.showWidgetName(children, true);
```
Comment (rahulramesha) [inline app/client/src/widgets/ModalWidget/widget/index.tsx:277]: NIT: Commented code can be removed if not required

**Comment by Sripriya93** [comment]
QA done for the fixes. @riodeuno

**Review by rahulramesha — APPROVED** [review]

File: app/client/src/widgets/ModalWidget/widget/index.tsx (hunk 2 above)
Comment (riodeuno) [reply app/client/src/widgets/ModalWidget/widget/index.tsx:277]: Thanks for noticing this @rahulramesha. I'll make a note and remove this in my next PR. This is a release blocker, and I'll have to wait for CI workflow again, if I make this fix in this PR. I'll fix it in this PR: https://github.com/appsmithorg/appsmith/pull/20118

**Pull request merged**